prisma-fmt/src/lib.rs

```diff
@@ -45,27 +46,49 @@ pub fn code_actions(schema: String, params: &str) -> String {
 }
 
 /// The two parameters are:
-/// - The Prisma schema to reformat, as a string.
+/// - The [`SchemaFileInput`] to reformat, as a string.
 /// - An LSP
 /// [DocumentFormattingParams](https://github.com/microsoft/language-server-protocol/blob/gh-pages/_specifications/specification-3-16.md#textDocument_formatting) object, as JSON.
 ///
 /// The function returns the formatted schema, as a string.
+/// If the schema or any of the provided parameters is invalid, the function returns the original schema.
+/// This function never panics.
 ///
 /// Of the DocumentFormattingParams, we only take into account tabSize, at the moment.
-pub fn format(schema: &str, params: &str) -> String {
+pub fn format(datamodel: String, params: &str) -> String {
+    let schema: SchemaFileInput = match serde_json::from_str(&datamodel) {
+        Ok(params) => params,
+        Err(_) => {
+            return datamodel;
+        }
+    };
+
     let params: lsp_types::DocumentFormattingParams = match serde_json::from_str(params) {
         Ok(params) => params,
-        Err(err) => {
-            warn!("Error parsing DocumentFormattingParams params: {}", err);
-            return schema.to_owned();
+        Err(_) => {
+            return datamodel;
         }
     };
 
-    psl::reformat(schema, params.options.tab_size as usize).unwrap_or_else(|| schema.to_owned())
+    let indent_width = params.options.tab_size as usize;
+
+    match schema {
```

Comment:
No strong reason, no. I've created a task to get rid of single versions of `reformat` and `validate` as you pointed out, so that I can get this integration version merged into `prisma/prisma` for  https://github.com/prisma/team-orm/issues/1096 👍🏻